internal/services/controller/controller_test.go

```diff
@@ -63,10 +67,22 @@ func TestMain(m *testing.M) {
 func TestController_HappyPath(t *testing.T) {
```

Comment:
suggestion: it could be good idea to update test name as well

With new test cases, it's not longer happy path testing